internal/kafka/internal/services/util_test.go

```diff
@@ -353,3 +395,41 @@ func Test_contains(t *testing.T) {
 		})
 	}
 }
+
+func Test_BuildCustomClaimCheck(t *testing.T) {
+	type args struct {
+		kafkaRequest      *dbapi.KafkaRequest
+		ssoconfigProvider keycloak.SSOProvider
+	}
+	kafkaRequest := buildKafkaDBApiRequest()
+	tests := []struct {
+		name                string
+		args                args
+		expectedCustomClaim string
+	}{
+		{
+			name: "Customclaimcheck without canary service account client ID - uses MAS SSO",
+			args: args{
+				kafkaRequest:      &kafkaRequest,
+				ssoconfigProvider: keycloak.MAS_SSO,
+			},
+			expectedCustomClaim: fmt.Sprintf("@.rh-org-id == '%s'|| @.org_id == '%s'", kafkaRequest.OrganisationId, kafkaRequest.OrganisationId),
+		},
+		{
+			name: "Customclaimcheck with canary service account client ID - uses REDHAT SSO",
+			args: args{
+				kafkaRequest:      &kafkaRequest,
+				ssoconfigProvider: keycloak.REDHAT_SSO,
+			},
+			expectedCustomClaim: fmt.Sprintf("@.rh-org-id == '%s'|| @.org_id == '%s' || @.clientId == '%s'", kafkaRequest.OrganisationId, kafkaRequest.OrganisationId, kafkaRequest.CanaryServiceAccountClientID),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			receivedBuiltCustomClaimCheck := BuildCustomClaimCheck(tt.args.kafkaRequest, tt.args.ssoconfigProvider)
+			if receivedBuiltCustomClaimCheck != tt.expectedCustomClaim {
+				t.Errorf("BuildCustomClaimCheck() = %v, want %v", receivedBuiltCustomClaimCheck, tt.expectedCustomClaim)
+			}
```

Comment:
Let's use gomega to perform the assertions.

Reply:
Thanks! should i update for other tests  as well in the file? a Few are not usng the gomega format, I will update for the other tests as well

Reply:
Thanks. That will be better. But if it starts to become too complex, maybe just focus on this one and the rest will be updated as we go.

Reply:
Done, updated the entire file to goemga for assertions